src/include/filepath.h

```diff
@@ -90,14 +92,20 @@ class filepath {
    * if we are fed a relative path as a string, either set ino=0 (strictly
    * relative) or 1 (absolute).  throw out any leading '/'.
    */
-  filepath(std::string_view s) { set_path(s); }
-  filepath(const char* s) { set_path(s); }
+  filepath(std::string_view s) {
+    set_path(s);
+  }
+  filepath(const char* s) {
+    set_path(s);
+  }
 
   void set_path(std::string_view s, inodeno_t b) {
+    ceph_assert(s.size() > 0);
```

Comment:
This is probably not correct, I think empty paths are permitted when specifying the inode.